src/NUnitFramework/framework/Assert.That.cs

```diff
@@ -50,23 +76,47 @@ static public void That(bool condition, string message, params object[] args)
         /// an <see cref="AssertionException"/>.
         /// </summary>
         /// <param name="condition">The evaluated condition</param>
-        static public void That(bool condition)
+        public static void That(bool condition)
         {
             Assert.That(condition, Is.True, null, null);
         }
 
         #endregion
 
         #region Lambda returning Boolean
+
 #if !NET_2_0
+        
+        /// <summary>
+        /// Asserts that a condition is true. If the condition is false the method throws
+        /// an <see cref="AssertionException"/>.
+        /// </summary> 
+        /// <param name="condition">A lambda that returns a Boolean</param>
+        /// <param name="getExceptionMessage">A function to build the message included with the Exception</param>
+        public static void That(Func<bool> condition, Func<ConstraintResult, string> getExceptionMessage)
```

Comment:
Short answer: it doesn't need to be there.  My original goal was to present an overload for a `Func<string>`, so this isn't really required.  I thought it was when making my original edits.

I think this overload came in when I made some of my original edits.  I was trying to make the code more DRY before, and I had refactored the various Assert.That methods so as to reuse as much code as possible.

Since the ConstraintResult class is internal, it (and all related overloads) should likely be removed.